torch/autograd/_functions/stochastic.py

```diff
@@ -30,9 +30,10 @@ def backward(self, reward):
         output_probs = probs.gather(1, samples)
         output_probs.add_(1e-6).reciprocal_()
         output_probs.neg_().mul_(reward)
-        # TODO: add batched index_add
-        for i in range(probs.size(0)):
-            grad_probs[i].index_add_(0, samples[i], output_probs[i])
+        # Fill in gradients
+        idx = grad_probs.new().resize_(grad_probs.size(0)).fill_(1)
+        idx = idx.cumsum(0) - 1
+        grad_probs[idx.long(), samples.t()] = output_probs.t()
```

Comment:
I don't think it's equivalent. The old version behaves differently in case of duplicated samples (they get added, not overwritten)